Automattic/wp-calypso
# Fix state not being up to date in domains step

Related to https://github.com/Automattic/wp-calypso/issues/94432
Fixes #94432

## Proposed Changes

* Domain's step is submitting empty data and therefore breaking the checkout, this PR addresses it. Like in [Plans step ](https://github.com/Automattic/wp-calypso/blob/trunk/client/landing/stepper/declarative-flow/internals/steps-repository/unified-plans/index.tsx#L68), we had to handle state changes a bit differently. This PR does the same for Domains.

## Why are these changes being made?


* Checkout was not working for new users because domains step was submitting empty data
* for new users. Check the video in the [issue.](https://github.com/Automattic/wp-calypso/issues/94432)

## Testing Instructions



* Using incognito, navigate to {live-link}/setup/onboarding
* Create a new user with email
* Pick a paid domain 
* Select a paid plan
* Checkout should work fine
* Finalize the purchase and make sure everything you selected is indeed purchased.
* Check free plans and free domains
* Check own domain
* Check choose domain later
* Check free plan together with paid plan


## Pre-merge Checklist
